code/variables.py

```diff
@@ -0,0 +1,95 @@
+        self.LONMIN = '-5.0' #min longitude; -ve values in the WH e.g. 59.8W = -59.8 -30
+        self.LONMAX = '9.0' #min longitude; -ve values in the WH e.g. 59.8W = -59.8  30
+        self.XRES = 4.0              #x direction spatial resolution in km
+        self.YRES = 4.0              #y direction spatial resolution in km
+        self.TRES = 1                #temporal resolution in hrs
+        self.LAT_DISTANCE = 111.0    #the avg distance in km for 1deg lat for the region being considered                     
+        self.LON_DISTANCE = 111.0    #the avg distance in km for 1deg lon for the region being considered                     
+        self.STRUCTURING_ELEMENT = [[0, 1, 0],
+                                    [1, 1, 1],
+                                    [0, 1, 0]
+                                    ] #the matrix for determining the pattern for the contiguous boxes and must                        
+#have same rank of the matrix it is being compared against
+#criteria for determining cloud elements and edges                                  
+        self.T_BB_MAX = 243  #warmest temp to allow (-30C to -55C according to Morel and Sensi 2002
+        self.T_BB_MIN = 218  #cooler temp for the center of the system
+        self.CONVECTIVE_FRACTION = 0.90 #the min temp/max temp that would be expected in a CE.. this is highly conservative (only a 10K difference)
+        self.MIN_MCS_DURATION = 3    #minimum time for a MCS to exist
+        self.AREA_MIN = 2400.0       #minimum area for CE criteria in km^2 according to Vila et al. (2008) is 2400
+        self.MIN_OVERLAP = 10000.00   #km^2  from Williams and Houze 1987, indir ref in Arnaud et al 1992                            
+    #---the MCC criteria
+        self.ECCENTRICITY_THRESHOLD_MAX = 1.0  #tending to 1 is a circle e.g. hurricane,
+        self.ECCENTRICITY_THRESHOLD_MIN = 0.70 #tending to 0 is a linear e.g. squall line
+        self.OUTER_CLOUD_SHIELD_AREA = 80000.0 #km^2
+        self.INNER_CLOUD_SHIELD_AREA = 30000.0 #km^2
+        self.OUTER_CLOUD_SHIELD_TEMPERATURE = 233 #in K
+        self.INNER_CLOUD_SHIELD_TEMPERATURE = 213 #in K                                                                 
+        self.MINIMUM_DURATION = 6  #min number of frames the MCC must exist for (assuming hrly frames, MCCs is 6hrs) 
+        self.MAXIMUM_DURATION = 24 #max number of framce the MCC can last for   
+        self.DIRS = {'mainDirStr': "/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/output/paperData", 'TRMMdirName':"/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/paperData/TRMM", 'CEoriDirName': "/Users/youssefbiaz/Documents/USC/,2015-3Fall'15/5CSCI401/grab-tag-graph/baselineTimings/paperData/MERG"}
+        self.startDateTime = "200609110000"
+        self.endDateTime = "200609121200"
+
+
+    def setupJSON(self):
+        data = None;
+        with open('../config.txt') as f:
```

Comment:
@youssefbiaz You will find that working with I/O and network stuff is full of exceptions and problems.  It's a good habit to get into that when you open a file (or network connection, or database connection, etc...) you wrap the code in a try/except block.  In this case you'd want to catch the IOError exception which the code will raise if config.txt is missing.  In this case we would want to warn the user that the config file is missing, then return them the object with the default settings so we don't crash the entire program.  If you do this with network or database connections, sometimes a couple of back-off and retries will make sense.